bson/bsoncodec/string_codec.go

```diff
@@ -30,6 +32,8 @@ var (
 )
 
 // NewStringCodec returns a StringCodec with options opts.
+//
+// Deprecated: Use bson.NewRegistry to get a registry with the StringCodec registered.
```

Comment:
No, we plan to remove all deprecated code in Go Driver 2.0. The difference is that sometimes we have an alternative to recommend and sometimes we are just dropping support for a particular behavior (typically because we believe it is unused). The messaging of "will not be supported in Go Driver 2.0" basically means "we're planning to remove this and don't have an alternative to offer".

If the current messaging is unclear, we could add the "will be removed in Go Driver 2.0" note to all deprecation messages. Thoughts?

As far as code with no suggested alternative, I think it's still valuable to add a notice that it's going away and should not be used.